google/cloud/storage/internal/xml_node.cc

```diff
@@ -0,0 +1,105 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/storage/internal/xml_node.h"
+#include "google/cloud/internal/absl_str_cat_quiet.h"
+#include "google/cloud/internal/absl_str_replace_quiet.h"
+#include "absl/strings/str_format.h"
+#include <stack>
+
+namespace google {
+namespace cloud {
+namespace storage_internal {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+
+namespace {
+
+std::string EscapeXmlString(std::string const& val) {
+  return absl::StrReplaceAll(val, {{"&", "&amp;"},
+                                   {"<", "&lt;"},
+                                   {">", "&gt;"},
+                                   {"\"", "&quot;"},
+                                   {"'", "&apos;"}});
+}
+
+std::string EscapeXmlText(std::string const& val) {
+  return absl::StrReplaceAll(val,
+                             {{"&", "&amp;"}, {"<", "&lt;"}, {">", "&gt;"}});
+}
```

Comment:
Given how these are used, perhaps better names would be `EscapeXmlTag()` and `EscapeXmlContent()` respectively.

Reply:
Done